### PR TITLE
SIP for Fields

### PR DESCRIPTION
**Pull request by kangrongji**
The only difference I've made is, the inverse operator and the inverse-is-inverse predicate are combined into one constructor. Then the SIP follows immediately.

### CODE REVIEW

**Comment by felixwellen** [comment]
The code looks all good to me!
Where is 'fix-whitespace' showing up in the history of your branch?

**Comment by kangrongji** [comment]
I have no idea. I cannot even find this `fix-whitespace` in my local repo. It seems automatically added by `[Merge branch 'field' of](https://github.com/agda/cubical/pull/825/commits/4baa151c722d904d8ceaa68734304d824af5ff30) https://github.com/kangrongji/cubical [into field](https://github.com/agda/cubical/pull/825/commits/4baa151c722d904d8ceaa68734304d824af5ff30)`.

**Comment by kangrongji** [comment]
It's OK now @felixwellen

File: Cubical/Algebra/Field/Base.agda [hunk 1]
```diff
@@ -145,17 +152,38 @@ _$_ : {R S : Field ℓ} → (φ : FieldHom R S) → (x : ⟨ R ⟩) → ⟨ S 
...
+  -- faster with some sharing
+  null = autoDUARel (𝒮-Univ _) (λ A → A)
+  bin = autoDUARel (𝒮-Univ _) (λ A → A → A → A)
+
+FieldPath : (R S : Field ℓ) → FieldEquiv R S ≃ (R ≡ S)
+FieldPath = ∫ 𝒮ᴰ-Field .UARel.ua
 
-  isProp→⊥ : ∀ {A : Type ℓ} → isProp (A → ⊥)
-  isProp→⊥ = isPropΠ λ _ → isProp⊥
+uaFieldRing : {A B : Field ℓ} → FieldEquiv A B → A ≡ B
```
Comment (felixwellen) [inline Cubical/Algebra/Field/Base.agda:188]: Why does this one have 'Ring' in its name?

Comment (kangrongji) [reply Cubical/Algebra/Field/Base.agda]: Eh... Embarrassing...

**Review by felixwellen — APPROVED** [review]

**Comment by felixwellen** [comment]
Thanks!

**Pull request merged**